ledger/src/transaction_address_lookup_table_scanner.rs

```diff
@@ -0,0 +1,64 @@
+use {
+    bincode::deserialize,
+    lazy_static::lazy_static,
+    solana_sdk::{
+        address_lookup_table::{self, instruction::ProgramInstruction},
+        pubkey::Pubkey,
+        sdk_ids::SDK_IDS,
+        transaction::SanitizedVersionedTransaction,
+    },
+    std::collections::HashSet,
+};
+
+lazy_static! {
+    static ref SDK_IDS_SET: HashSet<Pubkey> = SDK_IDS.iter().cloned().collect();
+}
+
+pub enum ScanResult {
+    /// The address lookup table program was not used
+    NotFound,
+    /// The address lookup table program was present, with non-native programs.
+    /// If instructions extending an account lookup table were found, the list
+    /// of Pubkeys will be returned. However, this may be incomplete due to the
+    /// possibility of non-native ixs CPIing to extend.
+    NonNativeUsed(Vec<Pubkey>),
+    /// The address lookup table program was present, with only native programs.
+    /// If instructions extending an account lookup table were found, the list
+    /// of Pubkeys will be returned.
+    NativeUsed(Vec<Pubkey>),
+}
+
+pub fn scan_transaction(transaction: &SanitizedVersionedTransaction) -> ScanResult {
+    // If the program is not present in the account keys, it was not used
+    if !transaction
+        .get_message()
+        .message
+        .static_account_keys()
+        .iter()
+        .any(address_lookup_table::program::check_id)
+    {
+        return ScanResult::NotFound;
+    }
```

Comment:
sadly, this isn't precise.. the alt could be loaded via another alt. this is okay as long as non-cpi ix isn't touching the alt-loaded alt program. cpi ix is allowed to execute alt-loaded on-chain programs.

Reply:
how about SanitizedTransaction construction as was before then only fall back to `static_account_keys()`?

this will reduce otherwise unavoidable false positive warnings greatly...

```rust
let account_keys = bank
    .fully_verify_transaction(transaction)
    .map(|tx| tx.message.account_keys()).unwrap_or_else(|| {
        possibly_incomplete = true;
        transaction.get_message().message.static_account_key()
    });
if !possibly_incompete &&
    account_keys.iter().any(address_lookup_table::program::check_id) {
    return ScanResult::NotFound;
}
...
```

Reply:
I did something similar, outside the scanning.
i.e. we only scan if we cannot verify.

However...it requires cloning the `tx` because `fully_verify_transaction` consumes.